src/runtime/internal/scheduler.ts

```diff
@@ -1,7 +1,7 @@
-import { run_all } from './utils';
-import { set_current_component } from './lifecycle';
+import { run_all, Queue } from './utils';
+import { get_current_component, set_current_component } from './lifecycle';
 
-export const dirty_components = [];
+export const dirty_components = new Queue<any>();
```

Comment:
is there a component type that could be used here rather than `any`?

Reply:
Good question. I'd like to know the answer myself. :-) The type specified here would need to be the same type that is used in `get_current_component` and `set_current_component` in `src/runtime/internal/lifecycle.ts`. Both of them are currently untyped, so Typescript assigns them a type of `any`. However, I tried doing `import type { SvelteComponent } from './Component'` and then making this a `Queue<SvelteComponent>`, and TypeScript was apparently happy with that.

So it looks like `SvelteComponent` would be the right type to put here. But there's also `SvelteComponentDev` and `SvelteComponentTyped` listed in `src/runtime/index.ts`, and I'm not 100% sure what the differences are. There's also a comment at the definition of SvelteComponentTyped pointing to https://github.com/microsoft/TypeScript/issues/41770 as the reason for its existence, so perhaps @dummdidumm would know more about this.

Reply:
Update: both SvelteComponentDev and SvelteComponentTyped are subclasses of SvelteComponent, so it looks like SvelteComponent might indeed be the right type to put in here. I'm still hesitant to make that change because `dirty_components` was previously untyped, and I don't know if making its type stricter than before might break anyone's existing Svelte code.